spec/unit/indirector/rest_spec.rb

```diff
@@ -353,7 +353,7 @@ def save_request(key, instance, options={})
       it "calls get on the connection" do
         request = find_request('foo bar')
 
-        connection.expects(:get).with("#{url_prefix}/test_model/foo%20bar?environment=production&", anything).returns(mock_response('200', 'response body'))
+        connection.expects(:get).with("#{url_prefix}/test_model/foo+bar?environment=production&", anything).returns(mock_response('200', 'response body'))
```

Comment:
Will this be backwards compatible if an older 4.x agent sends %20 encoded spaces to a 5.0 puppetserver and we unescape in JRuby?

Are there any puppetserver specific REST endpoints we need to worry about? Or will puppetserver/jetty decode both '+' and '%20' as a space? /cc @camlow325 @rlinehan